### DISCUSSION TITLE
GStreamer Integration: TLS/SSL support not available; install glib-networking

### DISCUSSION THREAD

**Discussion by MIDIculous** [ant-media/Ant-Media-Server#5774] [FAQs] [answered]
I'm trying to make a secure WebRTC connection using [GStreamer 1.22 on macOS](https://gstreamer.freedesktop.org/download/#macos), based on [this example](https://github.com/GStreamer/gst-examples/blob/master/webrtc/multiparty-sendrecv/gst/mp-webrtc-sendrecv.c). When trying to connect to the signaling server via wss://, I get the following error:

> HTTP/1.1 6 TLS/SSL support not available; install glib-networking

Should the example work out-of-the-box? Is `glib-networking` somewhere in `GStreamer.framework`, and just isn't being found?

Or, if I have to add it manually: How can I do so without creating system-wide dependencies? Everything needs to be self-contained in one folder, which can be moved and sent to customers. So using a package manager to install `glib-networking` system-wide won't do.

In `GStreamer.framework/Libraries/gio/modules/` there's only `libgioopenssl.so`. I've tried loading that manually using `g_module_open_full`, before the call to `gst_init_check`. Loading succeeds, but I still get the above error.
Any help would be appreciated.

**Comment by yashtandon113** [comment]
@MIDIculous 
Please check out the below blog post for WebRTC using Gstreamer and Ant Media.
https://antmedia.io/gstreamer-tutorial-how-to-publish-play-webrtc-streams/

I hope it helps.

**Reply by MIDIculous** [reply]
@yashtandon113 

Yes we know about this blog and this is the blog we are following to install. Perhaps I'm missing something, but the blog does not specifically answer the question we are asking. Can you point me as to where the blog answers the question?

**Comment by MIDIculous** [comment, marked as answer]
**UPDATED**: There is an answer [Here](https://gitlab.freedesktop.org/gstreamer/gstreamer-project/-/issues/114#note_2153872). It is a bug in Gstreamer that was updated 2 weeks ago, but no push was sent.
